File: Dec2020/28Dec/NumberOfLinesToWriteString.java
/*
Problem Name: 806. Number of Lines To Write String
Problem Link: https://leetcode.com/problems/number-of-lines-to-write-string/
*/
class Solution {
    public int[] numberOfLines(int[] widths, String s) {
        int lines = 1, width = 0;
        char c[] = s.toCharArray();
        for(int i=0;i<s.length();i++){
            if(width + widths[c[i]-'a'] <= 100)
                width += widths[c[i]-'a'];
            else{
                lines++;
                width = 0;
                i--;
            }
        }
        return new int[]{lines,width};
    }
}

File: Dec2020/28Dec/NumberOfLinesToWriteString.cpp
/*
Problem name: 806. Number of Lines To Write String
Problem link: https://leetcode.com/problems/number-of-lines-to-write-string/
*/

class Solution {
public:
    vector<int> numberOfLines(vector<int>& widths, string s) {
        vector<int> result;
        int line=1, pixels=0, i=0;
        while(i<s.length()){
            if(pixels + widths[s[i]-97] <= 100){
                pixels += widths[s[i]-97];
                i++;
            }
            else{
                line++;
                pixels=0;
            }
        }
        result.push_back(line);
        result.push_back(pixels);
        return result;
    }
};
